Lib/fontTools/varLib/__init__.py

```diff
@@ -503,22 +504,88 @@ def _add_gvar(font, model, master_ttfs, tolerance=0.5, optimize=True):
 
 			gvar.variations[glyph].append(var)
 
-def _add_cvar(font, model, master_ttfs, tolerance=0.5):
+def _remove_TTHinting(font):
+	for tag in ("cvar", "cvt ", "fpgm", "prep"):
+		if tag in font:
+			del font[tag]
+	for attr in ("maxZones", "maxTwilightPoints", "maxStorage", "maxFunctionDefs", "maxInstructionDefs", "maxStackElements", "maxSizeOfInstructions"):
+		setattr(font["maxp"], attr, 0)
+	glyf = font["glyf"]
+	for name in glyf.keys():
+		glyph = glyf[name]
+		if hasattr(glyph, "program"):
+			glyph.program = Program()
+	# TODO: Modify gasp table to deactivate gridfitting for all ranges?
+
+def _merge_TTHinting(font, model, master_ttfs, tolerance=0.5):
 
-	log.info("Generating cvar")
+	log.info("Merging TT hinting")
 	assert "cvar" not in font
+
+	# Check that the existing hinting is compatible
+
+	# fpgm and prep table
+
+	for tag in ("fpgm", "prep"):
+		all_pgms = [m[tag].program.getBytecode() for m in master_ttfs if tag in m]
```

Comment:
ttProgam.Program class has `__eq__` method so in theory you should be able to compare two program instances as such, without needing to manually do getBytecode and then compare the bytes. you tried?